src/services/storage/models/offer.model.ts

```diff
@@ -52,3 +51,37 @@ export default class Offer extends Model {
     return this.totalCapacity.minus(this.utilizedCapacity)
   }
 }
+
+/**
+ * This function generate nested query for aggregating an avg billing price for offer for specific currency
+ * @param sequelize
+ * @param currency
+ */
+export async function getBillingPriceAvgQuery (sequelize: Sequelize, currency: 'usd' | 'eur' | 'btc' = 'usd') {
+  if (!config.get('storage.tokens')) {
+    throw new Error('"storage.tokens" not exist in config')
+  }
+
+  const supportedTokens = Object.entries(config.get('storage.tokens'))
+  const rates = await Rate.findAll()
+  const toDollars = `${supportedTokens.reduce(
+    (acc, [tokenAddress, symbol]) => {
+      const rate: number = rates.find(r => r.token === symbol)?.[currency] || 0
+      return `${acc} \n
+        when token = ${sequelize.escape(tokenAddress)}
+        then cast(price as real) * ${sequelize.escape(rate)}`
+    },
+    ''
+  )}`
+  return literal(`(
+    SELECT (
+      ROUND(
+        (SUM(
+          case
+            ${toDollars}
+            else 0
+          end) / COUNT(*) * 1024 / period * (3600 * 24)
```

Comment:
Why `* 1024`? Maybe can you add parentheses to make more clear how is this calculated please? 😅

Reply:
I add comments. Basically it converts the avg billing price to price/gb/day